react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: ''
+    };
+    this.passwordErrors = {};
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+  }
+
+  checkPassword() {
+    this.passwordErrors =
+      this.state.value === ''
+        ? {
+            icon: 'fas fa-times',
+            messageContent: 'A password is required.'
+          }
+        : this.state.value.length < 8
+          ? {
+              icon: 'fas fa-times',
+              messageContent: 'Your password is too short.'
+            }
+          : {
+              icon: 'fas fa-check',
+              messageContent: ''
+            };
+  }
+
+  render() {
+    this.checkPassword();
```

Comment:
After changing `checkPassword` to return an object you can update this line as follows:

```JSX
const passwordErrors = this.checkPassword();
```